op-proposer/proposer/driver.go

```diff
@@ -468,20 +497,37 @@ func (l *L2OutputSubmitter) loopL2OO(ctx context.Context) {
 	}
 }
 
+// The loopDGF proposes a new output every proposal interval. It does _not_ query
+// the DGF for when to next propose, as the DGF doesn't have the concept of a
+// proposal interval, like in the L2OO case. For this reason, it has to keep track
+// of the interval itself, for which it uses an internal ticker.
 func (l *L2OutputSubmitter) loopDGF(ctx context.Context) {
+	defer l.Log.Info("loopDGF returning")
 	ticker := time.NewTicker(l.Cfg.ProposalInterval)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
-			blockNumber, err := l.FetchCurrentBlockNumber(ctx)
-			if err != nil {
-				break
+			// prioritize quit signal
+			select {
+			case <-l.done:
+				return
+			default:
 			}
 
-			output, shouldPropose, err := l.FetchOutput(ctx, blockNumber)
-			if err != nil || !shouldPropose {
-				break
+			var (
+				output *eth.OutputResponse
+				err    error
+			)
+			// A note on retrying: because the proposal interval is usually much
+			// larger than the interval at which to retry proposing on a failed attempt,
+			// we want to keep retrying getting the output proposal until we succeed.
+			for output == nil || err != nil {
+				output, err = l.FetchDGFOutput(ctx)
+				if err != nil {
+					l.Log.Warn("Error getting DGF output, retrying...", "err", err)
+					time.Sleep(l.Cfg.OutputRetryInterval)
```

Comment:
Shutdown will get stuck while this is happening. Should we be checking the quit signal in this inner loop?

Reply:
moved the check into the inner loop, at the start